chia/consensus/block_creation.py

```diff
@@ -215,14 +212,12 @@ def create_foliage(
         removals_root = removal_merkle_set.get_root()
 
         generator_hash = bytes32([0] * 32)
-        if solution_program is not None:
-            generator_hash = std_hash(solution_program)
+        if block_generator is not None:
+            generator_hash = std_hash(block_generator.program)
 
         generator_refs_hash = bytes32([1] * 32)
         if generator_block_heights_list not in (None, []):
-            generator_ref_list_bytes = b"".join(
-                [(i).to_bytes(4, byteorder="big") for i in generator_block_heights_list]
-            )
+            generator_ref_list_bytes = b"".join([i.to_bytes(4, byteorder="big") for i in generator_block_heights_list])
```

Comment:
Use `uint32` rather than this ad-hoc code.

Reply:
Thanks - `generator_block_heights_list` is a List[uint32].

What is the best way to hash over that in Python? I'm guessing we want the hash over the concat of the bytes, not e.g. hash(tuple(generator_block_heights_list))

Reply:
The `uint32` type responds to `bytes` so you can go `bytes(uint32(32769))` to get `b'\x00\x00\x80\x01'`, for example.

Reply:
Cool, then once it's in `bytes` land we have access to those methods. okay, let me take a look.

Reply:
bytes(i)

Reply:
Its the same as i.to_bytes(4, byteorder="big")